packages/template-retail-react-app/app/components/shared/ui/index.jsx

```diff
@@ -0,0 +1,107 @@
+export {Button} from './Button'
+export {ButtonGroup} from './ButtonGroup'
+export {Center} from './Center'
+export {ChakraProvider} from './ChakraProvider'
+export {Checkbox} from './Checkbox'
+export {CloseButton} from './CloseButton'
+export {Container} from './Container'
+export {Divider} from './Divider'
+export {Drawer} from './Drawer'
+export {DrawerBody} from './DrawerBody'
+export {DrawerCloseButton} from './DrawerCloseButton'
+export {DrawerContent} from './DrawerContent'
+export {DrawerFooter} from './DrawerFooter'
+export {DrawerHeader} from './DrawerHeader'
+export {DrawerOverlay} from './DrawerOverlay'
+export {Fade} from './Fade'
+export {Flex} from './Flex'
+export {FormControl} from './FormControl'
+export {FormErrorMessage} from './FormErrorMessage'
+export {FormLabel} from './FormLabel'
+export {Grid} from './Grid'
+export {GridItem} from './GridItem'
+export {HStack} from './HStack'
+export {Heading} from './Heading'
+export {Icon} from './Icon'
+export {IconButton} from './IconButton'
+export {Image} from './Image'
+export {Img} from './Img'
+export {Input} from './Input'
+export {InputGroup} from './InputGroup'
+export {InputLeftElement} from './InputLeftElement'
+export {InputRightElement} from './InputRightElement'
+export {Link} from './Link'
+export {List} from './List'
+export {ListItem} from './ListItem'
+export {Modal} from './Modal'
+export {ModalBody} from './ModalBody'
+export {ModalCloseButton} from './ModalCloseButton'
+export {ModalContent} from './ModalContent'
+export {ModalFooter} from './ModalFooter'
+export {ModalHeader} from './ModalHeader'
+export {ModalOverlay} from './ModalOverlay'
+export {Popover} from './Popover'
+export {PopoverArrow} from './PopoverArrow'
+export {PopoverBody} from './PopoverBody'
+export {PopoverCloseButton} from './PopoverCloseButton'
+export {PopoverContent} from './PopoverContent'
+export {PopoverFooter} from './PopoverFooter'
+export {PopoverHeader} from './PopoverHeader'
+export {PopoverTrigger} from './PopoverTrigger'
+export {Portal} from './Portal'
+export {Radio} from './Radio'
+export {RadioGroup} from './RadioGroup'
+export {Select} from './Select'
+export {SimpleGrid} from './SimpleGrid'
+export {Skeleton} from './Skeleton'
+export {Spacer} from './Spacer'
+export {Spinner} from './Spinner'
+export {Stack} from './Stack'
+export {StackDivider} from './StackDivider'
+export {StylesProvider} from './StylesProvider'
+export {Text} from './Text'
+export {Tooltip} from './Tooltip'
+export {VStack} from './VStack'
+export {Wrap} from './Wrap'
+export {WrapItem} from './WrapItem'
+export {createStylesContext} from './createStylesContext'
+export {extendTheme} from './extendTheme'
+export {useBreakpoint} from './useBreakpoint'
+export {useBreakpointValue} from './useBreakpointValue'
+export {useDisclosure} from './useDisclosure'
+export {useMediaQuery} from './useMediaQuery'
+export {useMultiStyleConfig} from './useMultiStyleConfig'
```

Comment:
How do people override the non visual components like `ChakraProvider` or `createStylesContext` or `useMultiStyleConfig`?

I guess if you want to replace Chakra, you'd override these using a noop function?

Reply:
Yes, exactly 💯

Reply:
Also remember that you can override `_app-config` and remove the chakra provider safely (then you wouldn't need these shared/ui components for non-visuals like providers), but I kept these in case there is some similar provider pattern with a library like material-ui.